doc/source/glossary.rst

```diff
@@ -2,6 +2,552 @@
+                  [ 9, 10, 11]]])
+
+          >>> a.shape
+          (2, 2, 3)
+
+      ``a`` is depicted as a two-element array whose elements are 2x3 vectors.
+      From this point of view, rows and columns are the final two axes,
+      respectively, in any shape.
+
+      This rule helps you anticipate how a vector will be printed, and
+      conversely how to find the index of any of the printed elements. For
+      instance, in the example, the last two values of 8's index must be 0 and
+      2. Since 8 appears in the second of the two 2x3's, the first index must
+      be 1:
+
+        >>> a[1,0,2]
+        8
+
+      A convenient way to count dimensions in a printed vector is to
+      count ``[`` symbols after the open-parenthesis. This is
+      useful in distinguishing, say, a (1,2,3) shape from a (2,3) shape:
+
+      ..  code:: python
+
+          >>> a.shape
+          (2, 3)
+          >>> a
+          array([[0, 1, 2],
+                 [3, 4, 5]])
+
+          >>> a.shape
+          (1, 2, 3)
+          >>> a
+          array([[[0, 1, 2],
+                  [3, 4, 5]]])
+
+
+    along an axis
+
+      The result of an operation along an axis X is a vector in which X
+      disappears. This can surprise new users expecting the opposite.
+
+      The operation can be visualized this way:
+
+      Imagine a slice of array ``a`` where axis X has a fixed index
+      and the other dimensions are left full (``:``).
+
+        >>> a.shape
+        (2,3,4)
+        >>> a[:,0,:].shape
+        (2,4)
+
+      The slice has ``a``'s shape with the X dimension deleted. Saying an
+      operation ``op`` is ``performed along X`` means that ``op`` takes as its
+      operands slices having every value of X:
+
+         >>> np.sum(a,axis=1) == a[:,0,:] + a[:,1,:] + a[:,2,:]
+         array([[ True,  True,  True,  True],
+                [ True,  True,  True,  True]])
+
+
+    array
+
+      Used synonymously in the NumPy docs with `ndarray`, NumPy's basic structure.
+      See :doc:`Array objects. <reference/arrays>`
+
+
+    array_like
+
+      Any `sequence <https://docs.python.org/3/glossary.html#term-sequence>`_
+      that can be interpreted as an ndarray.  In addition to ndarrays
+      this category includes lists (possibly nested and with
+      different element types), tuples, and scalars. Any argument accepted by
+      :doc:`numpy.array <reference/generated/numpy.array>`
+      is array_like.
+
+      ..  code::
+
+          >>> x = np.array([[1,2.0],[0,0],(1+1j,3.)])
+
+          >>> x
+          array([[1.+0.j, 2.+0.j],
+                 [0.+0.j, 0.+0.j],
+                 [1.+1.j, 3.+0.j]])
+
+
+    array scalar
+
+      For uniformity in handling operands, NumPy treats
+      a :doc:`scalar <reference/arrays.scalars>` as an array of zero dimension.
+
+
+    `attribute <https://docs.python.org/3/glossary.html#term-attribute>`_
+      \
+
+
+    .base
+
+      If an array does not own its memory, then its ``.base`` attribute
```

Comment:
```suggestion
      If an array does not own its memory, then its :doc:`.base <reference/generated/numpy.ndarray.base>` attribute
```
Might be worthwhile to link to the ``ndarray.base`` method here, 
similar to how ``array()`` was referenced in the ``array_like`` section.